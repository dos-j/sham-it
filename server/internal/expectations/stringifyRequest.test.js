const stringifyRequest = require("./stringifyRequest");

describe("unit: stringifyRequest", () => {
  test("it should stringify a request", () => {
    expect(
      stringifyRequest({
        request: {
          method: "GET",
          pathname: "/test"
        }
      })
    ).toMatchSnapshot();
  });

  test("it should include the headers if they are available", () => {
    expect(
      stringifyRequest({
        request: {
          method: "GET",
          pathname: "/test",
          headers: { Accept: "application/json" }
        }
      })
    ).toMatchSnapshot();
  });

  test("it should include a querystring if there are query parameters", () => {
    expect(
      stringifyRequest({
        request: {
          method: "POST",
          pathname: "/test",
          query: { top: 100, skip: 500 }
        }
      })
    ).toMatchSnapshot();
  });

  test("it should include a body if one was sent", () => {
    expect(
      stringifyRequest({
        request: {
          method: "POST",
          pathname: "/test",
          body: "hello world"
        }
      })
    ).toMatchSnapshot();
  });

  test("it should json serialize the body if it is an object", () => {
    expect(
      stringifyRequest({
        request: {
          method: "POST",
          pathname: "/test",
          headers: { "Content-Type": "application/json" },
          body: { a: 1 }
        }
      })
    ).toMatchSnapshot();
  });
});
