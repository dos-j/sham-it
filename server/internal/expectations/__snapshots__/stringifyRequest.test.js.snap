// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit: stringifyRequest it should include a body if one was sent 1`] = `
"
POST /test
  body:
      hello world"
`;

exports[`unit: stringifyRequest it should include a querystring if there are query parameters 1`] = `
"
POST /test?top=100&skip=500"
`;

exports[`unit: stringifyRequest it should include the headers if they are available 1`] = `
"
GET /test
  Accept: application/json"
`;

exports[`unit: stringifyRequest it should json serialize the body if it is an object 1`] = `
"
POST /test
  Content-Type: application/json
  body:
      {\\"a\\":1}"
`;

exports[`unit: stringifyRequest it should stringify a request 1`] = `
"
GET /test"
`;
